webapp/src/components/surveyPost/surveyPost.tsx

```diff
@@ -0,0 +1,243 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import {format} from 'date-fns';
+import React, {useCallback, useEffect, useMemo, useRef, useState} from 'react';
+import {useDebouncedCallback} from 'use-debounce';
+
+import Button from 'components/common/button/button';
+import LinearScaleQuestion from 'components/surveyPost/linearScaleQuestion/linearScaleQuestion';
+import TextQuestion from 'components/surveyPost/textQuestion/textQuestion';
+
+import './style.scss';
+
+import type {
+    CustomPostTypeComponentProps,
+    MattermostWindow,
+    SurveyResponse, UserSurvey,
+} from 'types/mattermost-webapp';
+
+function SurveyPost({post, isRHS}: CustomPostTypeComponentProps) {
+    const [survey, setSurvey] = useState<UserSurvey>();
+    const [disabled, setDisabled] = useState<boolean>(false);
+    const [expired, setExpired] = useState<boolean>(false);
+    const draftResponse = useRef<SurveyResponse>();
+    const [errorMessage, setErrorMessage] = useState<string>();
+
+    const linearScaleQuestionID = useRef<string>();
+
+    const fetchUserSurvey = async (surveyID: string) => {
```

Comment:
0/5, but perhaps it would be a good idea to factor out all the stuff around managing the survey state into a hook so that it could be tested independently of the component rendering?

Reply:
As in creating a custom hook to fetch the survey?

Reply:
Done. Created `useUserSurvey` hook. Please check it out. This is my first official custom hook 🎉 🎉 🎉